pkg/executor/aggregate/agg_spill.go

```diff
@@ -15,15 +15,287 @@
 package aggregate
 
 import (
+	"sync"
 	"sync/atomic"
 
+	"github.com/pingcap/tidb/pkg/executor/aggfuncs"
+	"github.com/pingcap/tidb/pkg/sessionctx"
+	"github.com/pingcap/tidb/pkg/util/chunk"
+	"github.com/pingcap/tidb/pkg/util/disk"
+	"github.com/pingcap/tidb/pkg/util/hack"
 	"github.com/pingcap/tidb/pkg/util/logutil"
 	"github.com/pingcap/tidb/pkg/util/memory"
 	"go.uber.org/zap"
 )
 
-// maxSpillTimes indicates how many times the data can spill at most.
-const maxSpillTimes = 10
+const (
+	// maxSpillTimes indicates how many times the data can spill at most.
+	maxSpillTimes       = 10
```

Comment:
Is this constraint still needed?

Reply:
> Is this constraint still needed?

yes, it's used for unparallel hash agg spill